### ISSUE TITLE
SDK Resolver: Allow versioned references to on-disk SDKs

### ISSUE THREAD

**Issue by dasMulli** [Microsoft/msbuild#3619]
I'd like if the `DefaultSdkResolver` would be able to resolve versioned SDK references from inside `MSBuildSdksPath` in addition to versionless references.
So `FooSdk/1.2.3` could be resolved from `{MSBuildSdksPath}/FooSdk/1.2.3/Sdk`.

This would help for integration testing custom SDKs where you could generate a version that won't end up on NuGet and make sure the project used for the test resolves to the version generated for the test.

Does this seem reasonable?
If so, I'd be happy to implement it.

**Comment by rainersigwald** [comment]
I remember we decided not to do this but I don't remember if it was "... for v1" or "ever". Right now I don't mind the idea, but I may have forgotten a good reason.

Biggest reason not to that I can think of is that we don't really want to encourage putting user stuff into MSBuild's installation directory, which is where MSBuildSdksPath lives (by default).

**Comment by dasMulli** [comment]
yeah it definitely isn't a common thing to do. I'm assembling a custom directory containing SDKs to set up an isolated integration test environment (lightweight sample: https://github.com/dasMulli/AssemblyInfoGenerationSdk / [these targets](https://github.com/dasMulli/AssemblyInfoGenerationSdk/blob/master/src/DasMulli.AssemblyInfoGeneration.Sdk/sdk-layout.targets)).
This requires test assets with verisonless references or removing the version from the reference.

**Comment by Nirmal4G** [comment]
We could add another SDK resolver with versioning similar to the default one and the NuGet one but for a custom SDKs path set in `global.json`. How about that?
